buildSrc/src/main/java/org/elasticsearch/gradle/test/RestApiSpecForTestingPlugin.java

```diff
@@ -0,0 +1,141 @@
+                        if (project.getPath().startsWith(":x-pack")) {
+                            Configuration xpackSpecConfig = project.getConfigurations().create("copyRestSpecXpack");
+                            Dependency xpackSpecDep = project.getDependencies()
+                                .project(Map.of("path", ":x-pack:plugin", "configuration", "restSpecXpack"));
+                            logger.info("X-pack rest specs for project [{}] will be copied to the test resources.", project.getPath());
+                            createCopyTask(project, xpackSpecConfig, xpackSpecDep, extension.getIncludesXpackSpec(), apiDir, false);
+                        }
+
+                        logger.info("Rest specs for project [{}] will be copied to the test resources.", project.getPath());
+                        createCopyTask(project, coreSpecConfig, coreSpecDep, extension.getIncludesCoreSpec(), apiDir, false);
+
+                    } else {
+                        // source the specs from the published jar - this is the path plugin developers
+                        logger.info(
+                            "Rest specs for project [{}] will be copied to the test resources from the published jar (version: [{}]).",
+                            project.getPath(),
+                            VersionProperties.getElasticsearch()
+                        );
+                        Dependency coreSpecFromJarDep = project.getDependencies()
+                            .create("org.elasticsearch:rest-api-spec:" + VersionProperties.getElasticsearch());
+
+                        createCopyTask(project, coreSpecConfig, coreSpecFromJarDep, extension.getIncludesCoreSpec(), "", true);
+                    }
+                } else {
+                    logger.info("Rest specs will be ignored for project [{}] since there are no REST tests", project.getPath());
+                    return;
+                }
+            } catch (Exception e) {
+                throw new IllegalStateException("Error configuring the rest-api-spec-for-testing plugin. This is likely a bug.", e);
+            }
+        });
+    }
+
+    private void createCopyTask(Project project, Configuration config, Dependency dep, List<String> includes, String into, boolean isJar) {
+        project.getDependencies().add(config.getName(), dep);
+        Copy copyTask = project.getTasks().create(config.getName() + "Task", Copy.class, copy -> {
+            copy.from(isJar ? project.zipTree(config.getSingleFile()) : config.getSingleFile());
+            copy.into(new File(getTestSourceSet(project).getOutput().getResourcesDir(), into));
+            if (includes.isEmpty()) {
+                copy.include(isJar ? apiDir + "/**" : "*/**");
+            } else {
+                includes.forEach(s -> copy.include(isJar ? apiDir + "/" + s + "*/**" : s + "*/**"));
+            }
+            copy.setIncludeEmptyDirs(false);
```

Comment:
Should be unnecessary since `false` is the default behavior here.

Reply:
done, and thanks!